backend/sirius-components-diagrams-layout/src/test/java/org/eclipse/sirius/components/diagrams/layout/services/DefaultTestDiagramDescriptionProvider.java

```diff
@@ -180,7 +184,27 @@ public DefaultTestDiagramDescriptionProvider(IObjectService objectService) {
         this.objectService = Objects.requireNonNull(objectService);
     }
 
+    private void populateTargetObjectIdToNodeId(List<Node> nodes, Map<String, String> targetObjectIdToNodeId) {
+        Optional<Node> optionalNode = Optional.empty();
+        List<Node> deeperNode = new ArrayList<>();
+
+        Iterator<Node> nodeIt = nodes.iterator();
+        while (optionalNode.isEmpty() && nodeIt.hasNext()) {
+            Node node = nodeIt.next();
+            targetObjectIdToNodeId.put(node.getTargetObjectId(), node.getId());
+            deeperNode.addAll(node.getChildNodes());
+            deeperNode.addAll(node.getBorderNodes());
+        }
```

Comment:
What is the role of this `optionalNode`? It is created empty and you are using it as a condition in your while but that's it. Are you sure that this code does what you think it does?

Reply:
Done